### ISSUE TITLE
Road to v0.2

### ISSUE THREAD

**Issue by matthewmturner** [datafusion-contrib/datafusion-dft#154]
This issue for setting the specific issues we would like completed before a v0.2 release.  My hope is that there can be a release by end of Sept / Beginning of October.

- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/137
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/138
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/128
- [ ] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/144 
- [x] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/125
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/153
- [x] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/148
- [x] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/158
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/159
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/166
- [x] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/167
- [x] (REQUIRED) https://github.com/datafusion-contrib/datafusion-dft/issues/170
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/168
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/173
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/175
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/177
- [x] (REQUIRED) Benchmark from CLI
- [x] (REQUIRED) Analyze from CLI
- [x] https://github.com/datafusion-contrib/datafusion-dft/issues/213
- [ ] https://github.com/datafusion-contrib/datafusion-dft/issues/224

**Comment by matthewmturner** [comment]
@alamb let me know if there is anything you would like to add

**Comment by alamb** [comment]
I think we should consider adding #148  as well.

**Comment by matthewmturner** [comment]
So I _think_ it turns out that the DDL issue I have is a bug with datafusion registering external tables to a schema.

I have this in my ~/.datafusion/.datafusionrc`

```
CREATE SCHEMA staging;

CREATE TABLE staging.foo AS VALUES (1);
CREATE EXTERNAL TABLE staging.min_aggs STORED AS PARQUET LOCATION 'ny2://atlas/sip/aggregates/minute_by_ticker_monthly_v2/year_month=2015-08/data.01.parquet';
```

But when I run `SHOW TABLES` I get the following:

<img width="1718" alt="image" src="https://github.com/user-attachments/assets/2cd8f262-73c0-4fb5-abf1-637c05564b17">

Notice how the schema isnt picked up for the the external table.  I dont see anything in the [docs](https://datafusion.apache.org/user-guide/sql/ddl.html#create-external-table) for `CREATE EXTERNAL TABLE` that would suggest this shouldnt work but maybe im misunderstanding.

@alamb do you have a view on this?  If you agree its a bug in datafusion then i can create a ticket there.

**Comment by alamb** [comment]
I agree it is a datafusion bug. I will file a ticket

**Comment by Kinrany** [comment]
Is there anything that the current trunk version does worse than v0.1?

**Comment by matthewmturner** [comment]
@Kinrany i dont think so although i havent used v0.1 in a long time (years).  i took a brief pause for december to work on some fun not work related project but im about to get back into dft in the beginning of the new year.  im going to review where things are again and potentially do a release early on before proceeding to the next round of features i have in mind.

**Comment by matthewmturner** [comment]
I was actually just poking around OpenDAL (specifically it's `ObjectStore` [implementation](https://github.com/apache/opendal/blob/main/integrations/object_store/README.md)) and it may actually be very easy to integrate hugging face - so i might see if it's as easy as im expecting and if so i will include that as well (basically the code in the example i linked but use the `services-huggingface` instead of `services-s3`).

**Comment by matthewmturner** [comment]
I was able to add the huggingface integration and ive started prepping for 0.2 release.  however, im currently pinned to a specific git commit for the hudi feature and you cant publish versions using that.  it looks like the hudi team is close to a new version but im not sure how long that will take.  

i might see if i can use the FFI table provider interface instead so i prevent these version issues from coming up in the future.

**Comment by matthewmturner** [comment]
I ended up removing the hudi feature for now so i could move forward with the 0.2 release - there are several other features other than the FFI table provider i would like to work on although i do plan to get back to that.
